x/attribute/keeper/keeper.go

```diff
@@ -600,3 +601,48 @@ func (k Keeper) ValidateExpirationDate(ctx sdk.Context, attr types.Attribute) er
 	}
 	return nil
 }
+
+// GetAccountData gets the value of the special accountdata attribute for the given address.
+func (k Keeper) GetAccountData(ctx sdk.Context, addr string) (string, error) {
+	attrs, err := k.GetAttributes(ctx, addr, types.AccountDataName)
+	if err != nil {
+		return "", fmt.Errorf("error finding %s for %q: %w", types.AccountDataName, addr, err)
+	}
+	// There should only ever be 0 or 1 entries. If there's more, just ignore the rest.
+	if len(attrs) == 0 {
+		return "", nil
+	}
+	return string(attrs[0].Value), nil
+}
+
+// SetAccountData sets/updates/deletes the value of the special accountdata attribute for a given address.
+// An error is only returned if the account data cannot be set as requested.
+func (k Keeper) SetAccountData(ctx sdk.Context, addr string, value string) error {
+	// Delete anything that might already be there.
+	existings, err := k.GetAttributes(ctx, addr, types.AccountDataName)
+	if err != nil {
+		return fmt.Errorf("could not look up existing %s for %q: %w", types.AccountDataName, addr, err)
+	}
+	if len(existings) > 0 {
+		err = k.DeleteAttribute(ctx, addr, types.AccountDataName, nil, k.modAddr)
+		if err != nil {
+			return fmt.Errorf("could not delete existing %s for %q: %w", types.AccountDataName, addr, err)
+		}
+	}
+
+	// Just leave it deleted if the new value is an empty string.
```

Comment:
I'm not sure how this is being used by the end user, but would they ever want an entry with an empty string? I'm guessing probably not since it would just be a waste of money.